src/lib/Liblog/chk_file_sec.c

```diff
@@ -452,3 +452,77 @@ tmp_file_sec(char *path, int isdir, int sticky, int disallow, int fullpath)
 	free(real);
 	return (rc);
 }
+
+/**
+ * @brief - This function takes an <program name> <args> as input in string format
+ *	    and returns the program name
+ *
+ * @return - char *
+ * @retval - NULL when no valid program name is found
+ * @retval - a newly allocated program name
+ *
+ * @NOTE - Caller holds the responsibility of freeing up the return value
+ */
+char *
+get_script_name(char *input) {
+	char * tok;
+	char *next_space;
+	int ret_fs;
+	int path_exists = 0;
+	char *prev_space = NULL;
+	int starts_with_quotes = 0;
+	char *delim = " ";
+	struct stat sbuf;
+
+	if (input == NULL)
+		return NULL;
+
+	/* if path starts with double quotes, skip it */
+	if (input[0] == '\"') {
+		input++;
+		starts_with_quotes = 1;
+	}
+
+	tok = strdup(input);
+	if (tok == NULL)
+		return NULL;
+
+	/* get rid of double quotes at the end */
+	if (starts_with_quotes && tok[strlen(tok)-1] == '\"')
+		tok[strlen(tok)-1] = '\0';
+
+	next_space = strpbrk(tok, delim);
+	for (; next_space != NULL; next_space = strpbrk(next_space, delim)) {
```

Comment:
You are initializing next_space outside of the for() loop.  Why not do it in the initialization portion?